src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -810,12 +811,15 @@ public virtual int Run(TextWriter consoleOutput, CancellationToken cancellationT
         /// <returns>A compilation that represents the original compilation with any additional, generated texts added to it.</returns>
         private protected (Compilation Compilation, GeneratorDriverTimingInfo DriverTimingInfo) RunGenerators(
             Compilation input,
+            string baseDirectory,
             ParseOptions parseOptions,
             ImmutableArray<ISourceGenerator> generators,
             AnalyzerConfigOptionsProvider analyzerConfigOptionsProvider,
             ImmutableArray<AdditionalText> additionalTexts,
             DiagnosticBag generatorDiagnostics)
         {
+            Debug.Assert(baseDirectory is not null);
```

Comment:
Think that we need a more descriptive name here. In the context of the compiler `baseDirectory` can have a variety of interpretations. Feel like we need to disambiguate this a bit.